dask_planner/src/sql.rs

```diff
@@ -152,6 +152,31 @@ impl ContextProvider for DaskSQLContext {
                 let rtf: ReturnTypeFunction = Arc::new(|_| Ok(Arc::new(DataType::Int64)));
                 return Some(Arc::new(ScalarUDF::new(name, &sig, &rtf, &fun)));
             }
+            "last_day" => {
+                let sig = Signature::variadic(
```

Comment:
Think we're going to want to amend the `Signature` variants we're using for these date functions to match up with the expected inputs; `variadic` implies that we could have an arbitrary amount of `Timestamps` passed through as input, when in this case we would only want one. For reference, [here's](https://docs.rs/datafusion-expr/13.0.0/datafusion_expr/struct.Signature.html) a list of the various signatures we could use.

I think that the signatures used here may be the cause for the datetime casting we later need to do on the Python end, as it looks like for the floor/ceil operations DataFusion is inserting some unnecessary casting to `Utf8` on the datetime columns.

Also want to note that this is a larger issue with almost all of our custom function signatures, and it would probably be worthwhile to comb through them and fix them where possible

Reply:
Interesting, I changed `variadic` from the custom functions for this PR. I've opened #868 for tracking the functions outside of this PR.